enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,915 @@
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --index=docker.io/my-namespace/community-operators --output=json`
+
+creates a new folder `community-operators` and downloads the package representations for the packages that are in the index
+
+```bash
+$ cd community-operators
+$ tree
+.
+├── etcd.json
+├── amqstreams.json
+├── image_sha256.json
+
+$ cat etcd.json
+[
+    {
+        "schema": "package.v1",
+        "name": "etcd",
+        "defaultChannel": "singlenamespace-alpha",
+        "icon": {
+            "base64data":"iVBORw0KGgoAAAANSUhEUgAAA.....",
+            "mediatype":"image/png"
+        },
+        "channels": ["alpha", "singlenamespace-alpha", "clusterwide-alpha"],
+        "description": "A message about etcd operator, a description of channels"
+    },
+    {
+        "schema": "bundle.v1", 
+        "name": "etcdoperator-community.v0.6.1",
+        "package": "etcd",
+        "image": "quay.io/operatorhubio/etcd:v0.6.1",
+        "version": "0.6.1",
+        "properties":[
+            {
+                "name": "olm.package",
+                "values": {
+                    "packageName": "etcd",
+                    "version": "0.6.1"
+                }
+            },
+            {
+                "name":"olm.gvk",
+                "values":{
+                    "group": "etcd.database.coreos.com", 
+                    "kind": "EtcdCluster", 
+                    "version": "v1beta2"
+                }
+            }
+        ],
+        "channels": [
+            {
+                "name": "alpha",
+                "upgradeInfo": {}
+            }
+        ],
+        "relatedImages": [
+            "quay.io/coreos/etcd-operator@sha256:bd944a211eaf8f31da5e6d69e8541e7cada8f16a9f7a5a570b22478997819943"
+        ]
+    },
+    {
+        "schema": "bundle.v1",
+        "name": "etcdoperator.v0.9.0",
+        "package": "etcd",
+        "image": "quay.io/operatorhubio/etcd:v0.9.0",
+        "version": "0.9.0",
+        "properties":[
+            {
+                "name": "olm.package",
+                "values": {
+                    "packageName": "etcd",
+                    "version": "0.9.0"
+                }
+            },
+            {
+                "name":"olm.gvk",
+                "values":{
+                    "group": "etcd.database.coreos.com", 
+                    "kind": "EtcdBackup", 
+                    "version": "v1beta2"
+                }
+            }
+        ],
+        "channels": [
+            {
+                "name": "singlenamespace-alpha",
+                "upgradeInfo": {
+                    "isReplacedBy": ["etcdoperator.v.0.9.2"]
+                }
+            },
+            {
+                "name": "clusterwide-alpha",
+                "upgradeInfo": {
+                    "isReplacedBy": ["etcdoperator.v0.9.2-clusterwide"]
```

Comment:
@bparees reverted to using `replaces` instead. With that change, I think it take care of the concerns you've stated above since it's the way the current state of the world is, and all components know how to handle skips/skipsrange with the information being persisted with the bundle info, and we can purge stale bundle blobs from the json file.